src/main/scala/io/getclump/ClumpSource.scala

```diff
@@ -3,10 +3,69 @@ package io.getclump
 import com.twitter.util.Future
 import scala.collection.generic.CanBuildFrom
 
-class ClumpSource[T, U] private[ClumpSource] (val functionIdentity: FunctionIdentity,
-                                              val fetch: Set[T] => Future[Map[T, U]],
-                                              val maxBatchSize: Int = Int.MaxValue,
-                                              val _maxRetries: PartialFunction[Throwable, Int] = PartialFunction.empty) {
+class ClumpSourceWithParam[T, U, P](val functionIdentity: FunctionIdentity,
+                                    val fetch: ((P, Set[T])) => Future[Map[T, U]],
+                                    val maxBatchSize: Int = Int.MaxValue,
+                                    val _maxRetries: PartialFunction[Throwable, Int] = PartialFunction.empty) {
+
+  def get(param: P, inputs: T*): Clump[List[U]] =
+    get(param, inputs.toList)
+
+  def get(param: P, inputs: List[T]): Clump[List[U]] =
+    Clump.collect(inputs.map(get(param, _)))
+
+  def get(param: P, input: T): Clump[U] = {
+    val parameterized: Set[T] => Future[Map[T, U]] = fetch(param, _)
+    val clumpSource: ClumpSource[T, U] = new ClumpSource[T, U](functionIdentity, parameterized, maxBatchSize, _maxRetries)
```

Comment:
I suspect having the `functionIdentity` at this level may be incorrect, maybe we should just call `ClumpSource.apply(parameterized)` here instead. Looking for guidance on this one